vcd/resource_vcd_ip_space_ip_allocation.go

```diff
@@ -0,0 +1,304 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdIpAllocation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdIpAllocationCreate,
+		ReadContext:   resourceVcdIpAllocationRead,
+		UpdateContext: resourceVcdIpAllocationUpdate,
+		DeleteContext: resourceVcdIpAllocationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdIpAllocationImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Org ID for IP Allocation",
+			},
+			"ip_space_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "IP Space ID for IP Allocation",
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Edge gateway name in which NAT Rule is located",
+				ValidateFunc: validation.StringInSlice([]string{"FLOATING_IP", "IP_PREFIX"}, false),
+			},
+			"usage_state": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Can be set to 'USED_MANUAL' to mark the IP Allocation for manual use",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Custom description can only be set when usage_state is set to 'USED_MANUAL'",
+			},
+			"prefix_length": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Optional:    true,
+				Computed:    true,
+				Description: "Required if 'type' is IP_PREFIX",
+			},
+			"ip_address": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "IP address or CIDR",
+			},
+			"used_by_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of entity that is using this allocation",
+			},
+			"allocation_date": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Allocation date in ISO 8601 format (e.g. 2023-06-07T09:57:58.721Z)",
+			},
+			"ip": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "IP address part",
+			},
```

Comment:
Yes - this is a "sugar" field. `ip_address` is the value that is returned from API - it will be an IP for `type=FLOATING_IP` or CIDR for `type=IP_PREFIX`. `ip` will always contain an IP - even in `CIDR` case